Feature/logical semaphore settings

bsanders opened the pull request:
This is the branch where I'm working on moving some attributes (and some non-attributes like the semaphore highlevel mark) to a settings file.

Note that these are no longer available in `list attr` commands, but *are* available in the profile/graph prefixed by `settings.`, and individual commands can access them just by importing `stack.settings`

bsanders committed:
FEATURE: Introduce a settings file for stacki.  Currently using it to set the maximum number of concurrent profile requests

bsanders committed:
BUGFIX: don't reimport stack - that breaks things because the namespacing is weird.  Instead, reload the module to get expected changes.

bsanders committed:
BUGFIX: Rename the Info_ attributes (now settings).  Make settings show up as 'settings.foo' in the graph to distinguish from attributes

bsanders committed:
FEATURE: move discovery base settings to settings.py

caladd reviewed (approved):
I like it, seems much cleaner. This give me a good foundation to build on when I get around to making `discover-nodes` a systemd service (writing the version parameters to that command to the settings file so the systemd daemon can read them when starting).

bsanders committed:
BUGFIX: remove /var/tmp/profile.{mutex,semaphore}.

These files are a redherring - because of systemd, the actual files that are used live under something horrific like: /var/tmp/systemd-private-{{BIGUUID}}-apache2.service-{{SMALL_HASH}}/tmp/profile.semaphore and change every apache restart.  That's ok because we generate the file in the correct path with the correct value from /opt/stack/etc/stacki.yml if it's missing.
